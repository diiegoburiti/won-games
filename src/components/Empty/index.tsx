import Image from 'next/image'
import Link from 'next/link'

import Button from 'components/Button'

import * as S from './styles'

export type EmptyProps = {
  title: string
  description: string
  hasLink?: boolean
}

const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <Image
      src="/img/empty.svg"
      alt="a gamer in a couch playing videogame"
      role="image"
      width={380}
      height={285}
    />

    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>

    {!!hasLink && (
      <Link href="/" passHref>
        <Button as="a">Go Back to Store </Button>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
