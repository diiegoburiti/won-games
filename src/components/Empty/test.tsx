import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

const props = {
  title: 'Nothing here',
  description: 'Some text here'
}

describe('<Empty />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Empty {...props} />)

    expect(
      screen.getByRole('image', {
        name: /a gamer in a couch playing videogame/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Nothing here/ })
    ).toBeInTheDocument()

    expect(screen.getByText(/Nothing here/i)).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render a Link when props hasLink is passed', () => {
    renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('link', { name: /Go Back to Store/i })
    ).toHaveAttribute('href', '/')
  })

  it('shouldnÂ´t render a Link when props hasLink is not passed', () => {
    renderWithTheme(<Empty {...props} />)

    expect(screen.queryByText(/Go Back to Store/i)).not.toBeInTheDocument()
  })
})
