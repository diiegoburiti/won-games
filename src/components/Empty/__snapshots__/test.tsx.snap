// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
<body>
  <div>
    <main
      class=""
    >
      <img
        alt="a gamer in a couch playing videogame"
        class=""
        role="image"
        src="/img/empty.svg"
      />
      <h2
        class=""
      >
        Nothing here
      </h2>
      <p
        class=""
      >
        Some text here
      </p>
    </main>
  </div>
</body>
`
